src/js/components/devices/filteritem.js

```diff
@@ -143,7 +149,7 @@ export default class FilterItem extends React.Component {
           options={filters.sort((a, b) => a.priority - b.priority)}
           renderInput={params => <TextField {...params} label="Attribute" style={textFieldStyle} />}
           key={reset}
-          value={key}
+          value={label}
```

Comment:
with the state stored label this leads to stale values in the autocomplete (e.g. if you clear the value) - doing it on render as you do in the [filters](https://github.com/mendersoftware/gui/pull/759/files#diff-1b8fef05bf9dba7c806bcefec3265fc7R145) should be enough here as well

Reply:
Done!